code/utils.py

```diff
@@ -0,0 +1,209 @@
+import yaml
```

Comment:
consider splitting up this utils file -- some of the functions have to do with df preprocessing, others have to do with location